oai_agents/common/learner.py

```diff
@@ -0,0 +1,83 @@
+class LearnerType:
+    '''
+    Different learners would receive differnt rewards from the base_overcooked_env.py.
+    The primary difference between learners are on the weights of personal and group reward.
+    Saboteur and Selfisher weight personal reward positively and group reward negatively.
+    Soloworker, weight individual reward positively, and group reward with a zero.
+    Collaborator, and Supporter weight both rewards positively.
+    If we grade how much they care about group: 
+    Supporter > Collaborator > Soloworker >> Selfisher > Saboteur.
+    If more details are desired, learner.py define these learners.
+    '''
+    SABOTEUR = "saboteur"
+    SELFISHER = "selfisher"
+    SOLOWORKER = "soloworker"
+    COLLABORATOR = "collaborator"
+    SUPPORTER = "supporter"
+
+class Learner:
+    def __new__(cls, learner_type: str):
```

Comment:
I don't think you need this function. `__init__` is enough for instance creation.